src/nitro/src/enclaves.rs

```diff
@@ -52,61 +69,59 @@ impl NitroEnclave {
             launcher.add_vcpu(None).map_err(NitroError::VcpuAdd)?;
         }
 
-        let sockaddr = VsockAddr::new(VMADDR_CID_PARENT, ENCLAVE_READY_VSOCK_PORT);
-        let listener = VsockListener::bind(&sockaddr).map_err(NitroError::HeartbeatBind)?;
-
         let cid = launcher
             .start(self.start_flags, None)
             .map_err(NitroError::VmStart)?;
 
-        // Safe to unwrap.
-        let cid: u32 = cid.try_into().unwrap();
+        Ok(cid.try_into().unwrap()) // Safe to unwrap.
+    }
 
+    fn poll(&self, listener: &VsockListener) -> Result<()> {
         let poll_timeout = PollTimeout::try_from((&self.image, self.mem_size_mib << 20))
             .map_err(NitroError::PollTimeoutCalculate)?;
 
-        enclave_check(listener, poll_timeout.into(), cid)?;
+        let mut poll_fds = [PollFd::new(listener.as_fd(), PollFlags::POLLIN)];
+        let result = poll(
+            &mut poll_fds,
+            NixPollTimeout::from(c_int::from(poll_timeout) as u16),
+        );
 
-        Ok(cid)
-    }
-
-    fn rootfs_archive(&self) -> Result<Vec<u8>> {
-        let mut builder = tar::Builder::new(Vec::new());
+        if result == Ok(0) {
+            return Err(NitroError::PollNoSelectedEvents);
+        }
 
-        builder
-            .append_dir_all("rootfs", self.rootfs.clone())
-            .unwrap();
+        if result != Ok(1) {
+            return Err(NitroError::PollMoreThanOneSelectedEvent);
+        }
 
-        Ok(builder.into_inner().unwrap())
+        Ok(())
```

Comment:
Instead of doing the individual checks on the result values you can do a match with a guard
```rust
        match result {
            Ok(0) => Err(NitroError::PollNoSelectedEvents),
            Ok(x) if x > 1 => Err(NitroError::PollMoreThanOneSelectedEvent),
            _ => Ok(())
        }
```